### PR TITLE
trackma: init at 0.8.4

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

https://github.com/z411/trackma

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by WeebSorceress** [comment]
Same as PR #176331, this was largely based on the flake pieces of @hlissner (https://github.com/hlissner/dotfiles/blob/master/packages/trackma.nix) and @danielphan2003 (https://github.com/danielphan2003/flk/blob/main/pkgs/applications/video/trackma/default.nix).

**Comment by danielphan2003** [comment]
Note that trackma-qt doesn't show anime cover image, and trackma-qt4 is broken.

**Comment by WeebSorceress** [comment]
> Note that trackma-qt doesn't show anime cover image, and trackma-qt4 is broken.

Yeah... I'm really having trouble packaging QT stuff (had a similar problem with https://github.com/NixOS/nixpkgs/pull/176331#discussion_r890616151). Should we not expose `trackma-qt` on `all-packages.nix` for now?

**Comment by danielphan2003** [comment]
> Yeah... I'm really having trouble packaging QT stuff (had a similar problem with [#176331 (comment)](https://github.com/NixOS/nixpkgs/pull/176331#discussion_r890616151)). Should we not expose `trackma-qt` on `all-packages.nix` for now?

Sure thing. I'm using the latest rev and QT is still buggy.

**Comment by WeebSorceress** [comment]
Result of `nixpkgs-review pr 179715` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>trackma</li>
    <li>trackma-curses</li>
    <li>trackma-gtk</li>
  </ul>
</details>

**Comment by WeebSorceress** [comment]
@danielphan2003 I removed `trackma-qt` for now... Once this PR gets merged I'm gonna do another one for [adl](https://github.com/RaitaroH/adl) so we complete the trinity.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Comment by WeebSorceress** [comment]
Result of `nixpkgs-review pr 179715` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>trackma</li>
    <li>trackma-curses</li>
    <li>trackma-gtk</li>
  </ul>
</details>

File: pkgs/tools/misc/trackma/default.nix
Comment (viraptor) [inline pkgs/tools/misc/trackma/default.nix:23]:
```suggestion
  nativeBuildInputs = lib.optionals withGtk [ wrapGAppsHook ];
```

File: pkgs/tools/misc/trackma/default.nix
Comment (viraptor) [inline pkgs/tools/misc/trackma/default.nix:25]:
```suggestion
  buildInputs = lib.optionals withGtk [ glib gobject-introspection gtk3 ];
```

File: pkgs/tools/misc/trackma/default.nix
Comment (AndersonTorres) [inline pkgs/tools/misc/trackma/default.nix:11]:
```suggestion
}:

```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]
Otaku invasion at nixpkgs? Ooooh noooo!

**Comment by toyo-chi** [comment]
> Note that trackma-qt doesn't show anime cover image

> I'm really having trouble packaging QT stuff

Packaging trackma for myself for a pretty long time and have no issues with Qt UI... are you intrigued?

**Comment by toyo-chi** [comment]
Well, still don't know how to easily and optimally share NixOS packages, so gist: https://gist.github.com/toyo-chi/d89e804c1e3e27341b2ccea835c017c8

Maybe not so pretty, but works for me!

**Comment by WeebSorceress** [comment]
> Just manually wrapping in preFixup. Maybe not so pretty, but works for me!

Hey @toyo-chi, sorry for taking that long to reply. Indeed, wrapping it in `preFixup` seems to solve the problem. I'm gonna push another PR fixing it soon. Thanks!

P.S. Feel free to add yourself as a maintainer to this package too. :wink:

**Comment by toyo-chi** [comment]
> I'm gonna push another PR fixing it soon. Thanks!

And please do not forget to enable [submodules](https://gist.github.com/toyo-chi/d89e804c1e3e27341b2ccea835c017c8#file-trackma-nix-L23), otherwise anime-relations module [will not be fetched](https://aur.archlinux.org/packages/trackma#comment-787846). And also please include icon and desktop files too.

> P.S. Feel free to add yourself as a maintainer to this package too.

I-i'm not ready for this, *senpai*!

**Pull request merged**
